### ISSUE TITLE
Seralizing short text id's in ENHO/ENHS might lead to problems

### ISSUE THREAD

**Issue by marcushoepfner** [larshp/abapGit#3428]
In ENHO/ENHS shorttext id's are serialized e.g. for BAdIs.
This might lead to problems since in on premise systems OTR is used as text repository (hence 32 characters OTR id's are serialized) and in Sap Cloud Platform enhancement text repository is used (which needs 20 characters id's).
Furthermore the id's might not exist in the target system.
What I want to say is, that they are actually kind of useless in the file and might lead to problems..

What about leaving them out and when importing check the enho/enhs in the system and take the ID's from the database. If the objects are new, create the ID's by just creating the objects without ID's. ID's are automatically determined by the system.

**Comment by larshp** [comment]
what is the "Sap Cloud Platform enhancement text repository" ? can you provide some context or examples?

**Comment by marcushoepfner** [comment]
Yes, it is ABAP in SAP cloud platform. We recently did a change and do not save the texts in OTR but enhancement text repo (just a table).
Enhancement Implementations (BAdI) and Enhancement Spots (BAdI) are supported soon.

**Comment by christianguenter2** [comment]
I noticed that there's also ```BADI_LONGTEXT_ID``` e.g. in structure ```enh_badi_data```. What about that?
![image](https://user-images.githubusercontent.com/17437789/83943646-14c1d300-a7fe-11ea-9061-65e53f36950e.png)

**Comment by marcushoepfner** [comment]
Yes, you are right. I did not implement handling for that yet, but I guess above mentioned approach would work. They are rarely used I guess.

**Comment by larshp** [comment]
@ThomasPloski do you know if this has been fixed?

**Comment by ThePlenkov** [comment]
Hi! I just wanted to tell that abapgit for ENHO doesn't work at all - pushing it in one system - breaks it in another. Diff shows only OTRs but seems like it complemetly breaks the implementation

**Comment by larshp** [comment]
examples welcome

**Comment by ThePlenkov** [comment]
well from diff perspective it's not visible - only uuids and texts are different:

![Image](https://github.com/user-attachments/assets/dcf867c3-1af4-4708-9e8c-0ebe2abc56d4)

But what actually happens:
- you pull such badi
-  immediately this implementation disappears from badi implementations
- this becomes also visible in a diff - like it becomes create instead of overwrite in abapgit UI
-  but even though - creation doesn't work because it says that implementation already exists


So it seems like a complex issue. To reproduce - just follow there steps:
- Create a new badi defintion
-  Create a new enhancement implementation

Move it to a new system via AG - notice that it will still show a diff

- now stage this diff back and pull  in a first system

Enhancement implementation will disappear

Probably it's not even related to OTRs
